src/main/java/org/firmata4j/firmata/transport/FirmataTransportInterface.java

```diff
@@ -0,0 +1,28 @@
+package org.firmata4j.firmata.transport;
+
+import java.io.IOException;
+
+/**
+ * @author akia
+ * @since 2018-01
+ */
+public interface FirmataTransportInterface {
+
+    /**
+     * start transport and initialize connector
+     */
+    void startTransport() throws IOException;
+
+    /**
+     * shutdown connector and stop the transport
+     */
+    void stopTransport() throws IOException;
+
+    /**
+     * send a message to device
+     *
+     * @param bytes data to send
+     */
+    void sendMessage(byte[] bytes) throws IOException;
+
```

Comment:
There should be a method to register a parser. That way the same transport may be used for different protocol versions.